substrate-node/pallets/pallet-tfgrid/src/node.rs

```diff
@@ -358,9 +353,9 @@ impl<T: Config> Clone for SerialNumber<T> {
 
 pub fn validate_serial_number(input: &[u8]) -> bool {
     input == DEFAULT_SERIAL_NUMBER
-        || input
-            .iter()
-            .all(|c| c.is_ascii_alphanumeric() || matches!(c, b'-' | b'_' | b'.'))
+        || input.iter().all(|c| {
+            c.is_ascii_alphanumeric() || c.is_ascii_whitespace() || matches!(c, b'-' | b'_' | b'.')
```

Comment:
Maybe we can check if input doesn't start (and end) with whitespace here to avoid weird cases like `" "` or `"    xxx  "`

Reply:
Well, if the manufactor decides to set this as the serial number it is out of our control. Zos will not set empty strings as well so that is good enough